parser-typechecker/src/Unison/Codebase/Serialization/V0.hs

```diff
@@ -324,6 +338,10 @@ putPattern putA p = case p of
   Pattern.EffectBind a r cid args k
     -> putWord8 9 *> putA a *> putReference r *> putLength cid
                   *> putFoldable args (putPattern putA) *> putPattern putA k
+  Pattern.SequenceLiteral a ps
+    -> putWord8 10 *> putA a *> putFoldable ps (putPattern putA)
+  Pattern.SequenceOp a l op r
```

Comment:
So, unless there's a super compelling reason (is there, here?), I'd just make the order of the fields match left to right order of the pattern. So `a`, then `l`, then `op`, then `r`.

Minor reason is serialization code should be as dumb as possible, using left-to-right order requires less thinking, less opportunity for errors, and is easier to inspect for correctness. Also, not for nothing, the deserialization case becomes simpler, can use applicative rather monadic syntax.